diff --git a/libfreefare/mifare_desfire.c b/libfreefare/mifare_desfire.c
index cb812ea..990baf6 100644
--- a/libfreefare/mifare_desfire.c
+++ b/libfreefare/mifare_desfire.c
@@ -157,7 +157,7 @@ static ssize_t	 read_data (MifareTag tag, uint8_t command, uint8_t file_no, off_
     do { \
 	static uint8_t __msg[MAX_FRAME_SIZE] = { 0x90, 0x00, 0x00, 0x00, 0x00, /* ..., */ 0x00 }; \
 	/*                                       CLA   INS   P1    P2    Lc    PAYLOAD    LE*/ \
-	static uint8_t __res[MAX_FRAME_SIZE]; \
+	static uint8_t __res[MAX_FRAME_SIZE+1]; \
 	size_t __len = 5; \
 	errno = 0; \
 	if (!msg) return errno = EINVAL, -1; \
@@ -173,7 +173,7 @@ static ssize_t	 read_data (MifareTag tag, uint8_t command, uint8_t file_no, off_
 	MIFARE_DESFIRE (tag)->last_pcd_error = OPERATION_OK; \
 	DEBUG_XFER (__msg, __len, "===> "); \
 	int _res; \
-	if ((_res = nfc_initiator_transceive_bytes (tag->device, __msg, __len, __res, __##res##_size, 0)) < 0) { \
+	if ((_res = nfc_initiator_transceive_bytes (tag->device, __msg, __len, __res, __##res##_size+1, 0)) < 0) { \
 	    return errno = EIO, -1; \
 	} \
 	__##res##_n = _res; \
